src/ripple/shamap/impl/SHAMapInnerNode.cpp

```diff
@@ -74,50 +74,34 @@ SHAMapInnerNode::getChildIndex(int i) const
 std::shared_ptr<SHAMapTreeNode>
 SHAMapInnerNode::clone(std::uint32_t cowid) const
 {
-    auto const branchCount = getBranchCount();
-    auto const thisIsSparse = !hashesAndChildren_.isDense();
-    auto p = std::make_shared<SHAMapInnerNode>(cowid, branchCount);
+    // We are cloning it for writing, and who knows what it'll be used for,
```

Comment:
You could probably use some depth based heuristics here to try not to overallocate - but given what I /assume/ is the lifecycle of a node - not sure how much of a win that would be.